python/src/trezorlib/firmware/core.py

```diff
@@ -0,0 +1,203 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2022 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import hashlib
+import typing as t
+from copy import copy
+from enum import Enum
+
+import construct as c
+from construct_classes import Struct, subcon
+
+from .. import cosi
+from ..tools import EnumAdapter, TupleAdapter
+from . import consts, util
+from .vendor import VendorHeader
+
+__all__ = [
```

Comment:
They're here because of the star imports in `__init__.py`. If I didn't declare `__all__`, then I'd also be importing the local symbols from here (like `t`, `c`, `copy` etc.) which would mess up my global namespace and in some weird case could even make behavior dependent on the _order_ of the star imports.

I could instead avoid the star imports in `__init__` and list every symbol there, but this is nicer because the `__all__` is closer to the place where we might be adding or removing the exported symbols.